lib/Runtime/Library/amd64/JavascriptFunctionA.S

```diff
@@ -2,147 +2,122 @@
 // Copyright (C) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
 //-------------------------------------------------------------------------------------------------------
+
 .intel_syntax noprefix
+#include "unixasmmacros.inc"
 
-#include "pal/inc/unixasmmacros.inc"
-        
 .extern _ZN2Js18JavascriptFunction13DeferredParseEPPNS_14ScriptFunctionE
 
-.global C_FUNC(amd64_CallFunction)
 .global	_ZN2Js18JavascriptFunction20DeferredParsingThunkEPNS_16RecyclableObjectENS_8CallInfoEz
 
 #ifndef __APPLE__
-.type C_FUNC(amd64_CallFunction), @function
 .type _ZN2Js18JavascriptFunction20DeferredParsingThunkEPNS_16RecyclableObjectENS_8CallInfoEz, @function
 #endif
-        
-.text
-C_FUNC(amd64_CallFunction):
-        // Var amd64_CallFunction(RecyclableObject *function,
-        //	JavascriptMethod entryPoint, CallInfo callInfo, uint argc, Var *argv) ;
 
-        // Save existing rbp
-        push rbp
+//------------------------------------------------------------------------------
+// Invoke
+//
+//      JavascriptMethod(RecyclableObject* function, CallInfo callInfo, ...)
+//
+// with a custom calling convention in order to support JavascriptStackWalker
+// and RUNTIME_ARGUMENTS. JavascriptMethod entry stack layout:
+//
+//      [Return Address]
+//      [function]              == RDI
+//      [callInfo]              == RSI
+//      [arg0]
+//      [arg1]
+//      ...
+//------------------------------------------------------------------------------
+NESTED_ENTRY amd64_CallFunction, _TEXT, NoHandler
+        // Var amd64_CallFunction(
+        //      RecyclableObject *function,             (rdi)
+        //	JavascriptMethod entryPoint,            (rsi)
+        //      CallInfo callInfo,                      (rdx)
+        //      uint argc,                              (rcx)
+        //      Var *argv) ;                            (r8)
+
+        push_nonvol_reg rbp             // push rbp and adjust CFA offset
```

Comment:
I would add a little about CFA = Canonical Frame as described by the DWARF spec (and maybe a link to it) since most of the team would probably not be familiar with it